Sources/Routing/Protocols/NavigationPathManageable.swift

```diff
@@ -10,28 +10,27 @@ import SwiftUI
 public protocol NavigationPathManageable: ObservableObject {
     associatedtype Destination: ViewDisplayable
 
-    var path: NavigationPath { get set }
+    var path: [Destination] { get set }
 
-    func pop(_ count: Int)
+    func popTo(_ destination: Destination)
     func popToRoot()
     func push(_ destination: Destination)
     func push(_ destinations: [Destination])
 }
 
 extension NavigationPathManageable {
-    /// Removes the specified number of destinations from the navigation path
-    /// - Parameter count: The number of views to be popped from the navigation path
-    public func pop(_ count: Int = 1) {
-        guard count <= path.count else {
-            path = .init()
-            return
-        }
-        path.removeLast(count)
+    // Function to pop the navigation stack up to a specified destination
+    public func popTo(_ destination: Destination) {
```

Comment:
how will this handle if the destination exists multiple times in the stack?

Reply:
this will pop to the last specified destination in the stack

Reply:
Can you write tests for this & other cases? (e.g. if it doesn't exist. if it's on top of the stack)

Reply:
I made a new pull request that has all the changes include unit test